CppImport/src/macro/StaticStuff.cpp

```diff
@@ -0,0 +1,358 @@
+			else
+				Q_ASSERT(false);
+		}
+		else if (auto ooField = DCast<OOModel::Field>(node))
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->fields()->append(ooField);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->fields()->append(ooField);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->fields()->append(ooField);
+			else
+				Q_ASSERT(false);
+		}
+		else if (auto ooVarDecl = DCast<OOModel::VariableDeclaration>(node))
+		{
+			if (auto context = DCast<OOModel::Method>(declaration))
+				context->items()->append(new OOModel::ExpressionStatement(
+													 new OOModel::VariableDeclarationExpression(ooVarDecl)));
+			else
+				Q_ASSERT(false);
+		}
+		else if (auto ooMethod = DCast<OOModel::Method>(node))
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->methods()->append(ooMethod);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->methods()->append(ooMethod);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->methods()->append(ooMethod);
+			else
+				Q_ASSERT(false);
+		}
+		else
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else if (auto context = DCast<OOModel::Method>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else
+				Q_ASSERT(false);
+		}
+	}
+	else
+		Q_ASSERT(false && "not implemented");
+}
+
+QVector<Model::Node*> StaticStuff::topLevelNodes(QVector<Model::Node*> input)
+{
+	QSet<Model::Node*> topLevel;
+	for (auto node : input) topLevel.insert(node);
+
+	for (auto node : input)
+		for (auto other : input)
+			if (node != other)
+				if (node->isAncestorOf(other))
+					topLevel.remove(other);
+
+	QVector<Model::Node*> result;
+	for (auto node : topLevel) result.append(node);
+	return result;
+}
+
+void StaticStuff::buildMappingInfo(Model::Node* node, QList<Model::Node*>* info)
+{
+	info->push_back(node);
+
+	for (auto child : node->children())
+		buildMappingInfo(child, info);
+}
+
+void StaticStuff::useMappingInfo(Model::Node* node, QList<Model::Node*>* info, NodeMapping* mapping)
+{
+	mapping->add(info->front(), node);
+	info->pop_front();
+
+	for (auto child : node->children())
+		useMappingInfo(child, info, mapping);
+}
+
+OOModel::MetaCallExpression* StaticStuff::containsMetaCall(Model::Node* node)
+{
+	if (auto metaCall = DCast<OOModel::MetaCallExpression>(node))
+		return metaCall;
+
+	for (auto child : node->children())
+		if (auto metaCall = containsMetaCall(child))
+			return metaCall;
+
+	return nullptr;
+}
+
+bool StaticStuff::stringMatches(QString regex, QString value)
```

Comment:
I would call this `matchesRegex`. Altough I'm not srue we should even have this function. How often do we call this? Can't we just use the QRegularExpression directly at the call site?